src/proxy/processors/push-action/pullRemote.ts

```diff
@@ -2,9 +2,163 @@ import { Action, Step } from '../../actions';
 import fs from 'fs';
 import git from 'isomorphic-git';
 import gitHttpClient from 'isomorphic-git/http/node';
+import path from 'path';
+import os from 'os';
+import { simpleGit } from 'simple-git';
 
 const dir = './.remote';
 
+type BasicCredentials = {
+  username: string;
+  password: string;
+};
+
+type CloneResult = {
+  command: string;
+  strategy: Action['pullAuthStrategy'];
+};
+
+const ensureDirectory = (targetPath: string) => {
+  if (!fs.existsSync(targetPath)) {
+    fs.mkdirSync(targetPath, { recursive: true, mode: 0o755 });
+  }
+};
+
+const decodeBasicAuth = (authHeader?: string): BasicCredentials | null => {
+  if (!authHeader) {
+    return null;
+  }
+
+  const [scheme, encoded] = authHeader.split(' ');
+  if (!scheme || !encoded || scheme.toLowerCase() !== 'basic') {
+    throw new Error('Invalid Authorization header format');
+  }
+
+  const credentials = Buffer.from(encoded, 'base64').toString();
+  const separatorIndex = credentials.indexOf(':');
+  if (separatorIndex === -1) {
+    throw new Error('Invalid Authorization header credentials');
+  }
+
+  return {
+    username: credentials.slice(0, separatorIndex),
+    password: credentials.slice(separatorIndex + 1),
+  };
+};
+
+const buildSSHCloneUrl = (remoteUrl: string): string => {
+  const parsed = new URL(remoteUrl);
+  const repoPath = parsed.pathname.replace(/^\//, '');
+  return `git@${parsed.hostname}:${repoPath}`;
+};
+
+const cleanupTempDir = async (tempDir: string) => {
+  try {
+    await fs.promises.rm(tempDir, { recursive: true, force: true });
```

Comment:
I'm wondering about the use of `fs.promises` along with `fs.existsSync`/`fs.mkdirSync`. Can/should these be consistent?

Reply:
Fixed in https://github.com/finos/git-proxy/pull/987/commits/f1b4ddbd61c0728182bf68cded780d3781e6ec2b